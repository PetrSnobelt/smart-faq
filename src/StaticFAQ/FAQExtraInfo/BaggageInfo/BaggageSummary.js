// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import css from 'styled-jsx/css';

import BaggageLoader from './BaggageLoader';
import BaggageDescriptionFragment from './BaggageDescription';
import replaceWithCurrentDomain from '../../../helpers/replaceWithCurrentDomain';
import type { BaggageSummary as BaggageSummaryProps } from './__generated__/BaggageSummary.graphql';

type Props = {|
  data: ?BaggageSummaryProps,
  mmbUrl: string,
|};

const styles = css`
  .moreInfo {
    padding-left: 24px;
    padding-bottom: 24px;
  }
  .moreInfo a {
    color: #00a991;
    font-size: 12px;
    text-decoration: none;
    font-weight: 500;
  }
  hr.separationLine {
    height: 1px;
    background-color: #e8edf1;
    border: none;
  }
`;

export const RawBaggageSummary = ({ data, mmbUrl }: Props) => {
  return data ? (
    <React.Fragment>
      {data.map((baggage, i) => (
        /* eslint-disable react/no-array-index-key*/
        <BaggageDescriptionFragment key={i} data={baggage} />
      ))}
      <div className="moreInfo">
        <a
          target="_blank"
          rel="noopener noreferrer"
          href={replaceWithCurrentDomain(mmbUrl)}
        >
          More Info
        </a>
      </div>
      <style jsx>{styles}</style>
    </React.Fragment>
  ) : (
    <React.Fragment>
      <hr className="separationLine" />
      <BaggageLoader />
      <style jsx>{styles}</style>
    </React.Fragment>
  );
};

export default createFragmentContainer(
  RawBaggageSummary,
  graphql`
    fragment BaggageSummary on BookingBaggage @relay(plural: true) {
      ...BaggageDescription
    }
  `,
);
