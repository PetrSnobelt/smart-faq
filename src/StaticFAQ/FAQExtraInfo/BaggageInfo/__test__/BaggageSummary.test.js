// @flow

import * as React from 'react';
import { render, shallow } from 'enzyme';

import { RawBaggageSummary } from '../BaggageSummary';

const mockRefType: any = null;

describe('BaggageSummary', () => {
  it('loading screen renders correctly', () => {
    expect(
      render(<RawBaggageSummary data={null} mmbUrl="https://example.com" />),
    ).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const data = [
      {
        $refType: mockRefType,
        $fragmentRefs: mockRefType,
      },
      {
        $refType: mockRefType,
        $fragmentRefs: mockRefType,
      },
    ];

    expect(
      shallow(<RawBaggageSummary data={data} mmbUrl="https://example.com" />),
    ).toMatchSnapshot();
  });
});
