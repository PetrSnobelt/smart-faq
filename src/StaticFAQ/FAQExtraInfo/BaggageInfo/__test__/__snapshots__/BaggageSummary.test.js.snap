// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaggageSummary loading screen renders correctly 1`] = `
Array [
  <hr
    class="jsx-3371679792 separationLine"
  />,
  <div
    class="jsx-624713577 baggageRow"
  >
    <svg
      preserveAspectRatio="xMidYMid meet"
      version="1.1"
      viewBox="0 0 500 30"
    >
      <rect
        clip-path="url(#y7ayapd935l)"
        height="30"
        style="fill:url(#ttdlsn52gti)"
        width="500"
        x="0"
        y="0"
      />
      <defs>
        <clipPath
          id="y7ayapd935l"
        >
          <rect
            class="jsx-624713577"
            height="18"
            rx="0"
            ry="0"
            width="25"
            x="0"
            y="12"
          />
          <rect
            class="jsx-624713577"
            height="18"
            rx="0"
            ry="0"
            width="130"
            x="40"
            y="12"
          />
          <rect
            class="jsx-624713577"
            height="18"
            rx="0"
            ry="0"
            width="70"
            x="470"
            y="12"
          />
        </clipPath>
        <linearGradient
          id="ttdlsn52gti"
        >
          <stop
            offset="0%"
            stop-color="#e8edf1"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              repeatCount="indefinite"
              values="-2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="#d9dee2"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              repeatCount="indefinite"
              values="-1.5; 1.5"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#e8edf1"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              repeatCount="indefinite"
              values="-1; 2"
            />
          </stop>
        </linearGradient>
      </defs>
    </svg>
  </div>,
]
`;

exports[`BaggageSummary renders correctly 1`] = `
<React.Fragment>
  <Relay(BaggageDescription)
    data={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
      }
    }
    key="0"
  />
  <Relay(BaggageDescription)
    data={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
      }
    }
    key="1"
  />
  <div
    className="jsx-3371679792 moreInfo"
  >
    <a
      className="jsx-3371679792"
      href="null"
      rel="noopener noreferrer"
      target="_blank"
    >
      More Info
    </a>
  </div>
  <JSXStyle
    css=".moreInfo.jsx-3371679792{padding-left:24px;padding-bottom:24px;}.moreInfo.jsx-3371679792 a.jsx-3371679792{color:#00a991;font-size:12px;-webkit-text-decoration:none;text-decoration:none;font-weight:500;}hr.separationLine.jsx-3371679792{height:1px;background-color:#e8edf1;border:none;}"
    styleId="3371679792"
  />
</React.Fragment>
`;
